src/pudl/transform/ferc1.py

```diff
@@ -2869,66 +2869,25 @@ def process_dbf(self, raw_dbf):
         raw_dbf = super().process_dbf(raw_dbf)
         return raw_dbf
 
-    def assign_record_id(
-        self, df: pd.DataFrame, source_ferc1: SourceFerc1
-    ) -> pd.DataFrame:
-        """Add a column identifying the original source record for each row.
-
-        Generate the ``record_id`` column in similar fashion to
-        :meth:`Ferc1AbstractTableTransformer.assign_record_id`, but use the table name
-        from the ``dbf_to_xbrl.csv`` during the :meth:`align_row_numbers_dbf` for the
-        DBF side of the table because this DBF input came from two seperate tables.
-
-        Args:
-            df: table to assign ``record_id`` to
-            table_name: name of table
-            source_ferc1: data source of raw ferc1 database.
+    def source_table_id(
```

Comment:
I'm guessing this is returning a `str` or a `pd.Series` so that the return value can be used identically in a dataframe assignment elsewhere, with the string getting broadcast across all rows, and the Series assumed (guaranteed?) to have the same indexing as the dataframe whose column it is being used to populate? How do we know that those indexes will always line up / how do we check whether they line up and throw an error if they don't?

Reply:
If the series is returned and it has a different index, the assign will only assign values to rows that share indexes. I can check to ensure that there are no null `source_table_id` within `assign_record_id`. But if they got reindexed i don't really know how I would be able to test that. I added this, but I think in general with passing around series and assigning them as columns you need to be careful with your indexes.

```
    if df.source_table_id.isnull().any():
            raise ValueError(
                f"{self.table_id.value}: Null source_table_id's were found where none "
                "were expected."
            )
```

Reply:
Good enough! At least we'll know if something is wrong.